lib/app/providers.dart

```diff
@@ -0,0 +1,17 @@
+import 'package:firebase_auth/firebase_auth.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'package:logger/logger.dart';
+import 'package:starter_architecture_flutter_firebase/services/firestore_database.dart';
+
+final firebaseAuthProvider =
+    Provider<FirebaseAuth>((ref) => FirebaseAuth.instance);
+
+final databaseProvider =
+    ScopedProvider<FirestoreDatabase>((ref) => throw UnimplementedError());
```

Comment:
tip:

There's a shorthand for that:

```dart
final databaseProvider = ScopedProvider<FirestoreDatabase>(null);
```

Reply:
But considering what you are doing in `AuthWidget`, you may want to remove the AuthWidget from `ScopedProvider` and instead do:

```dart
final databaseProvider = Provider<FirestoreDatabase>((ref) {
  final auth = ref.watch(authStateChangesProvider);

  if (auth.data?.value?.user?.id != null) {
    return FirestoreDatabase(auth.data.value.user.id);
  }
  return null;  
});
```

Reply:
Thanks for the suggestions. I fixed all the providers and it works much better now :)